src/llama-graph.cpp

```diff
@@ -803,6 +803,10 @@ ggml_tensor * llm_graph_context::build_ffn(
 
     if (down) {
         cur = build_lora_mm(down, cur);
+        if (arch == LLM_ARCH_GLM4) {
+            // GLM4 seems to have precision issues in F16
```

Comment:
```suggestion
            // GLM4 seems to have numerical issues with half-precision accumulators
```

This is what I would expect the problem to be, without actually having looked into it.

Reply:
Good call, considering the code path for FP16 w/ FP32 out does work